homeassistant/components/mealie/services.py

```diff
@@ -73,27 +73,25 @@
 )
 
 SERVICE_SET_MEALPLAN = "set_mealplan"
-SERVICE_SET_MEALPLAN_SCHEMA = vol.Any(
+SERVICE_SET_MEALPLAN_SCHEMA = vol.All(
     vol.Schema(
         {
             vol.Required(ATTR_CONFIG_ENTRY_ID): str,
             vol.Required(ATTR_DATE): cv.date,
             vol.Required(ATTR_ENTRY_TYPE): vol.In(
                 [x.lower() for x in MealplanEntryType]
             ),
-            vol.Required(ATTR_RECIPE_ID): str,
+            vol.Exclusive(ATTR_RECIPE_ID, "abc"): str,
+            vol.Exclusive(ATTR_NOTE_TITLE, "abc"): str,
+            vol.Optional(ATTR_NOTE_TEXT): str,
```

Comment:
Hmm, so what I want to achieve is that either `recipe_id` or `note_title` is set. And only if `note_title` is set, `note_text` is allowed.

Reply:
You can consider using a `vol.Any(vol.Schema({…}), vol.Schema({…}))` but the error message will suck. Search `key_value_schemas` as the solution that we made :)

Reply:
Oh wait no, that's if you validate based on the value of a key, not existence of a key. We do that too with `determine_script_action` but it's a bit more hacky.